src/parser/lval.js

```diff
@@ -378,4 +380,18 @@ export default class LValParser extends NodeUtils {
       }
     }
   }
+
+  checkToRestConversion(node: SpreadElement): void {
+    const validArgumentTypes = [
+      "Identifier",
+      "MemberExpression",
+      "AssignmentPattern",
```

Comment:
Oh, yeah, Ive seen this as allowed so Ive added it as valid. Gonna remove this and add a test